ssm.tf

```diff
@@ -0,0 +1,19 @@
+resource "aws_ssm_parameter" "pem" {
+  count = local.ssm_enabled ? 1 : 0
+
+  name  = format(var.secret_path_format, module.this.name, "pem")
+  type  = "SecureString"
```

Comment:
@korenyoni  unless I'm mistaken, 
if `type  = "SecureString"`, a KMS Key ID/ARR is required, no?
It says "optional" here https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ssm_parameter#key_id, but it's optional if 
`type  = "String"` but required if `type  = "SecureString"`

Reply:
This is an instance of the AWS Provider documentation being inadequate. [ssm_parameter.key_id](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ssm_parameter#key_id) doesn't mention what KMS key is used when `key_id` is not supplied, whereas [secretsmanager_secret.kms_key_id](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/secretsmanager_secret#kms_key_id) does.

The AWS Provider [performs a test](https://github.com/hashicorp/terraform-provider-aws/blob/e31620b7229900e77f60e16485cbd1b36809eaff/aws/resource_aws_ssm_parameter_test.go#L612-L621) on `aws_ssm_parameter` with `type = "SecureString"` and no `key_id` supplied:


```go
func testAccAWSSSMParameterSecureConfig(rName string, value string) string {
	return fmt.Sprintf(`
resource "aws_ssm_parameter" "secret_test" {
  name        = "test_secure_parameter-%[1]s"
  description = "description for parameter %[1]s"
  type        = "SecureString"
  value       = "%[2]s"
}
`, rName, value)
}
```

```go
...
		Steps: []resource.TestStep{
			{
				Config: testAccAWSSSMParameterSecureConfig(name, "secret"),
				Check: resource.ComposeTestCheckFunc(
					testAccCheckAWSSSMParameterExists(resourceName, &param),
					resource.TestCheckResourceAttr(resourceName, "value", "secret"),
					resource.TestCheckResourceAttr(resourceName, "type", "SecureString"),
					resource.TestCheckResourceAttr(resourceName, "key_id", "alias/aws/ssm"), // Default SSM key id
				),
			}
...
```

So, inadequate information from the AWS Provider documentation, but it is fully the case that `key_id` is not required.

Maybe we should make this a variable?

Reply:
yes, we provide a var for `key_id` in many other modules